edisgo/flex_opt/reinforce_grid.py

```diff
@@ -737,6 +766,7 @@ def reinforce():
             "reinforcement."
         )
         selected_timesteps = converging_timesteps
+        troubleshooting_mode = troubleshooting_mode_set
         reinforce()
 
     # Run reinforcement for time steps that did not converge after initial reinforcement
```

Comment:
I don't understand what you mean here, sorry. The if statement above this line is already [if not converging_timesteps.empty](https://github.com/openego/eDisGo/blob/wp_flex_changed_reinforcement/edisgo/flex_opt/reinforce_grid.py#L763).

Reply:
I mean to add the respective statement here: https://github.com/openego/eDisGo/blob/70c3264caa39bbe875231c9b88f8415b4c182d01/edisgo/flex_opt/reinforce_grid.py#L773

It is just in the case that no timesteps converged, it is unnecessary to run lines 774-780, because the timesteps did not converge before and the grid did not change. So it is certain that the timesteps will not converge again. So instead of "if not non_converging_timesteps.empty:", I would write "if not non_converging_timesteps.empty and not converging_timesteps.empty:"
Does that make sense?

Reply:
But that could only happen, if no time steps were selected for the grid reinforcement in the first place, in which case one wouldn't get to this part of the code due to the return statement before.